### ISSUE TITLE
Add Astro example

### ISSUE THREAD

**Issue by pomber** [code-hike/codehike#255]
Astro 1.0 will have better [support for MDX](https://docs.astro.build/en/guides/integrations-guide/mdx/). It looks like it adds some custom stuff on top of MDX so I'm not sure Code Hike will still work.

**Comment by carvinlo** [comment]
Hi, thanks for this library! Bug When I try to use [`CH.Section`](https://codehike.org/docs/ch-section) component it throws an error.

**Comment by pomber** [comment]
Yes, the Astro integration is still buggy, mostly because of the `client` directive.

**Comment by may17** [comment]
Is this a problem astro needs to solve?

**Comment by pomber** [comment]
> Is this a problem astro needs to solve?

Not sure. I guess other remark-plugins that create client side components will have the same problem, so it would be nice to have some way to default to `client:load` in some cases.

I may be able to add a workaround in Code Hike but it will have to wait until next month at least.

**Comment by pomber** [comment]
Found another problem, [Astro converts children to static HTML](https://docs.astro.build/en/core-concepts/framework-components/#nesting-framework-components), and Code Hike needs to process them as objects.

**Comment by pomber** [comment]
We won't have full support for Astro in the near future. I tried, but there were at least 4 different problems blocking the integration.

**Comment by codersjj** [comment]
Is Astro v2 supported now?

**Comment by pomber** [comment]
> Is Astro v2 supported now?

It isn't, but there's now an experimental flag in Astro that may help in the future. It's still not working as expected thought. [Read more here](https://github.com/withastro/roadmap/discussions/465#discussioncomment-6898167).

**Comment by mbleigh** [comment]
What's the current state of Astro support with Code Hike? My personal blog is on Astro and I would love to be able to write posts that leverage Code Hike.

**Comment by pomber** [comment]
> What's the current state of Astro support with Code Hike?

The TL;DR is: Astro has its own flavor of MDX, Code Hike needs React MDX. 

[This proposal to allow frameworks to write Astro pages](https://github.com/withastro/roadmap/discussions/750) may solve the issue, but doesn't look like it's going to happen.

**Comment by divyanshu013** [comment]
Hi @pomber, I was following the thread [here](https://github.com/withastro/roadmap/discussions/465#discussioncomment-9499763). You mentioned about an adapter post v1 release. Does it look feasible in codehike now?
